Sources/SettingsFeature/Settings.swift

```diff
@@ -189,116 +193,133 @@ public struct Settings: ReducerProtocol {
 
   public init() {}
 
+  private enum CancelID {
+    case paymentObserver
+    case updateRemoveSettings
+  }
+
   public var body: some ReducerProtocol<State, Action> {
     CombineReducers {
       BindingReducer()
-      Reduce { state, action in
-        enum CancelID {
-          case paymentObserver
-          case updateRemoveSettings
-        }
-
-        switch action {
-        case .binding(\.$developer.currentBaseUrl):
-          return .fireAndForget { [url = state.developer.currentBaseUrl.url] in
-            await self.apiClient.setBaseUrl(url)
-            await self.apiClient.logout()
-          }
-
-        case .binding(\.$enableNotifications):
-          guard
-            state.enableNotifications,
-            let userNotificationSettings = state.userNotificationSettings
-          else {
-            // TODO: API request to opt out of all notifications
-            state.enableNotifications = false
-            return .none
+        .onChange(of: \.developer.currentBaseUrl.url) { _, url in
```

Comment:
The `onChange` stuff for bindings did get pretty gnarly, but not sure if there is a better way to do it.

Reply:
On the other hand, we've introduced isowords bugs in the past from nested binding state...